borsh-rs/borsh/src/de/mod.rs

```diff
@@ -410,6 +410,12 @@ impl BorshDeserialize for Box<[u8]> {
     }
 }
 
+impl<T: BorshDeserialize> BorshDeserialize for Box<T> {
```

Comment:
hm, curious how Box<[u8]> above doesn't compete with this given now there is also [T] now. But if it works, that's fine.

Reply:
Because `[T]` does not implement `BorshDeserialize` trait.